resources/user-agents/browsers/uc-browser/uc-web-8.json

```diff
@@ -172,27 +198,112 @@
           "match": "UCWEB/*(#DEVICE#*) U2/* UCBrowser/#MAJORVER#.#MINORVER#*",
           "engine": "U2",
           "devices": {
+            "BlackBerry": "Blackberry Blackberry",
             "BlackBerry*8520/": "Blackberry 8520",
             "BlackBerry*8530/": "Blackberry 8530",
             "BlackBerry*9320/": "Blackberry 9320",
-            "BlackBerry*9700/": "Blackberry 9700",
-            "BlackBerry": "Blackberry Blackberry"
+            "BlackBerry*9700/": "Blackberry 9700"
           },
           "platforms": [
             "RIMOS"
           ]
         },
         {
-          "match": "Mozilla/5.0 (#PLATFORM#GT-I9100 Build/*) applewebkit* (*khtml*like*gecko*) UCBrowser/#MAJORVER#.#MINORVER#* Mobile Safari/*",
+          "match": "Mozilla/5.0 (Java;*#DEVICE#) UCBrowser#MAJORVER#.#MINORVER#*UCWEB*Mobile*UNTRUSTED/*",
+          "devices": {
+            "samsung-gt-s5263": "Samsung GT-S5263"
+          },
+          "platforms": [
+            "JAVA"
+          ]
+        },
+        {
+          "match": "Mozilla/5.0 (Java;*#DEVICE#) applewebkit* (*khtml*like*gecko*) UCBrowser/#MAJORVER#.#MINORVER#*UCWEB*Mobile*",
+          "devices": {
+            "nokiae63-2": "Nokia E63",
+            "nokiae72-1": "Nokia E72"
+          },
+          "platforms": [
+            "SymbianOS"
+          ]
+        },
+        {
+          "match": "Mozilla/5.0 (Java;*) applewebkit* (*khtml*like*gecko*) UCBrowser/#MAJORVER#.#MINORVER#*UCWEB*Mobile*",
+          "device": "general Mobile Phone",
+          "platforms": [
+            "JAVA"
+          ]
+        },
+        {
+          "match": "Mozilla/5.0 (Java;*#DEVICE#) UCBrowser#MAJORVER#.#MINORVER#*UCWEB*Mobile*",
+          "devices": {
+            "nokiae71-1": "Nokia E71-1",
+            "Nokia8600 Luna": "Nokia 8600"
+          },
+          "platforms": [
+            "Series40"
+          ]
+        },
+        {
+          "match": "Mozilla/5.0 (#PLATFORM##DEVICE# Build/*) applewebkit* (*khtml*like*gecko*) UCBrowser/#MAJORVER#.#MINORVER#* Mobile Safari/*",
           "engine": "WebKit",
-          "device": "Samsung GT-I9100",
+          "devices": {
+            "GT-I9100": "Samsung GT-I9100"
+          },
           "platforms": [
-            "Android_4_0", "Android_4_1"
+            "Android_4_1", "Android_4_0"
           ]
         },
         {
-          "match": "Mozilla/5.0 (#PLATFORM#GT-I9100G Build/*) applewebkit* (*khtml*like*gecko*) UCBrowser/#MAJORVER#.#MINORVER#* Mobile Safari/*",
-          "device": "Samsung GT-I9100G",
+          "match": "Mozilla/5.0 (S60V* #DEVICE#) applewebkit* (*khtml*like*gecko*) UCBrowser/#MAJORVER#.#MINORVER#* Mobile*",
```

Comment:
For 8 and 9 I added a lot of devices to detect a number of patterns I found in our logs + other test suites.